specifyweb/frontend/js_src/lib/components/Statistics/index.tsx

```diff
@@ -0,0 +1,599 @@
+      className={{
+        container: dialogClassNames.wideContainer,
+      }}
+      header={statsText('chooseStatistics')}
+      onClose={handleClose}
+    >
+      <div>
+        <H3>{statsText('selectCustomStatistics')}</H3>
+        {Array.isArray(queries) && (
+          <QueryList
+            getQuerySelectCallback={(query) => () => {
+              handleAdd({ type: 'CustomStat', queryId: query.id });
+              handleClose();
+            }}
+            isReadOnly
+            queries={queries}
+          />
+        )}
+      </div>
+      <div>
+        {defaultStatsAddLeft.length > 0 && (
+          <div>
+            <H3>{statsText('selectDefaultStatistics')}</H3>
+            {defaultStatsAddLeft.map((defaultLayoutItem, index) => (
+              <div key={index}>
+                <H3>{defaultLayoutItem.label}</H3>
+                <div>
+                  <CategoriesBoxes
+                    goToEditingState={(): void => {}}
+                    isEditing={false}
+                    pageLayout={defaultLayoutItem}
+                    statsSpec={statsSpec}
+                    onClick={(item): void => {
+                      handleAdd(item);
+                      handleClose();
+                    }}
+                    onRemove={undefined}
+                  />
+                </div>
+              </div>
+            ))}
+          </div>
+        )}
+      </div>
+    </Dialog>
+  ) : null;
+}
+
+function CategoriesBoxes({
+  pageLayout,
+  statsSpec,
+  isEditing,
+  goToEditingState,
+  onClick: handleClick,
+  onRemove: handleRemove,
+}: {
+  readonly pageLayout: {
+    readonly label: string;
+    readonly categories: RA<{
+      readonly label: string;
+      readonly items: RA<CustomStat | DefaultStat>;
+    }>;
+  };
+  readonly statsSpec: IR<
+    IR<{
+      readonly label: string;
+      readonly items: StatCategoryReturn;
+    }>
+  >;
+  readonly isEditing: boolean;
+  readonly goToEditingState: (categoryIndex: number) => void;
+  readonly onClick: ((item: CustomStat | DefaultStat) => void) | undefined;
+  readonly onRemove:
+    | ((categoryIndex: number, itemIndex: number) => void)
+    | undefined;
+}): JSX.Element {
+  return (
+    <>
+      {pageLayout.categories.map(({ label, items }, categoryIndex) => (
+        <div
+          className="flex h-auto max-h-80 flex-col content-center rounded border-[1px] border-black bg-white p-4"
+          key={categoryIndex}
+        >
+          <H3 className="font-bold">{label}</H3>
+          <div className="overflow-auto pr-4">
+            {items?.map((item, itemIndex) => {
+              return item.type === 'DefaultStat' ? (
+                <DefaultStatItem
+                  categoryName={item.categoryName}
+                  itemName={item.itemName}
+                  pageName={item.pageName}
+                  key={itemIndex}
+                  statsSpec={statsSpec}
+                  onClick={
+                    handleClick !== undefined
```

Comment:
No negated conditions. Instead, swap th branches or check for `typeof handleClick === 'function'`